src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -1706,7 +1706,7 @@ void GcInfoDecoder::ReportRegisterToGC( // ARM64
     pCallBack(hCallBack, pObjRef, gcFlags DAC_ARG(DacSlotLocation(regNum, 0, false)));
 }
 
-#ifdef TARGET_UNIX
+#if defined(TARGET_UNIX) && !defined(FEATURE_REDHAWK)
```

Comment:
Nit: Could you please make the same change on ARM version of this function too? (It would be needed if we ever enabled ARM.)